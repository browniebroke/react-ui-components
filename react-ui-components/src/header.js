import React from 'react'
import styled from 'styled-components'
import Container from './container'

const SiteNavBar = styled.nav`
  background-color: ${(props) => props.theme.colors.primary};
  padding: ${(props) => props.theme.spacings[1]}
    ${(props) => props.theme.spacings[2]};
`

const HeaderContainer = styled(Container)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: ${(props) => props.theme.spacings[1]} 0;

  > a {
    display: flex;
    align-items: center;
  }
`

const Header = ({ children }) => (
  <header>
    <SiteNavBar>
      <HeaderContainer>{children}</HeaderContainer>
    </SiteNavBar>
  </header>
)

export default Header
