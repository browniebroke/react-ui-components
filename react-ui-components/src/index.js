export ExternalLink from './external-link'
export ListInline from './list-inline'

export styledComponentsTheme from './styled-components-theme'
export typography from './typography'

export { typographyTheme } from './typography'
