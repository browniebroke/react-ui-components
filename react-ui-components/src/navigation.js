import React from 'react'
import styled from 'styled-components'

const NavigationList = styled.ul`
  list-style: none;
  display: flex;
  flex-direction: ${(props) => (props.direction ? props.direction : 'column')};
  padding: 0;
`

const NavItem = styled.li`
  padding: ${(props) => props.theme.spacings[1]};

  a,
  a:hover {
    text-decoration: none;
  }
`

const Navigation = ({ children, direction = 'column' }) => (
  <NavigationList direction={direction}>
    {React.Children.map(children, (child) => (
      <NavItem>{child}</NavItem>
    ))}
  </NavigationList>
)

export default Navigation
