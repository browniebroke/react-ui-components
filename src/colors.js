import Color from 'color'

const getColor = (props, colorName) => {
  const colorValue = props.theme.colors[colorName]
  return Color(colorValue).hex()
}

const getColorDark = (props, colorName) => {
  const colorValue = props.theme.colors[colorName]
  return Color(colorValue).darken(0.2).hex()
}

export const getPrimaryColor = (props) => getColor(props, 'primary')
export const getPrimaryColorDark = (props) => getColorDark(props, 'primary')
export const getSecondaryColor = (props) => getColor(props, 'secondary')
export const getSecondaryColorDark = (props) => getColorDark(props, 'secondary')
export const getNavbarColor = (props) => getColor(props, 'navbar')
export const getNavbarColorDark = (props) => getColorDark(props, 'navbar')
export const getNavbarTextColor = (props) => getColor(props, 'navbarText')
export const getNavbarTextColorDark = (props) =>
  getColorDark(props, 'navbarText')

const mixAlertColor = (colorHex, factor) => {
  const baseColor = factor > 0 ? Color('#000000') : Color('#ffffff')
  return baseColor.mix(Color(colorHex), Math.abs(factor) * 0.08).hex()
}

export const getAlertBackgroundColor = (props) => {
  return mixAlertColor(props.theme.colors[`${props.alertType}`], -5)
}

export const getAlertTextColor = (props) => {
  return mixAlertColor(props.theme.colors[`${props.alertType}`], 6)
}
