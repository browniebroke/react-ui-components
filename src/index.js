export Alert from './alert'
export Col from './column'
export Container from './container'
export ContentWrapper from './content-wrapper'
export ExternalLink from './external-link'
export GlobalStyles from './global-styles'
export Header from './header'
export ResponsiveIframeWrapper from './iframe'
export ListInline from './list-inline'
export Navigation from './navigation'
export Row from './row'

export styledComponentsTheme from './styled-components-theme'
export typography from './typography'

export { typographyTheme } from './typography'
