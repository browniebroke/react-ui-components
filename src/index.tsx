export { default as Alert } from './alert'
export { default as Col } from './column'
export { default as Container } from './container'
export { default as ContentWrapper } from './content-wrapper'
export { default as ExternalLink } from './external-link'
export { default as GlobalStyles } from './global-styles'
export { default as Header } from './header'
export { default as ResponsiveIframeWrapper } from './iframe'
export { default as ListInline } from './list-inline'
export { default as Navigation } from './navigation'
export { default as Row } from './row'

export { default as styledComponentsTheme } from './styled-components-theme'
export { default as typography } from './typography'

export { typographyTheme } from './typography'
