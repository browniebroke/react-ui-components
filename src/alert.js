import styled from 'styled-components'
import PropTypes from 'prop-types'

import { getAlertBackgroundColor, getAlertTextColor } from './colors'

/**
 * Alert component
 */
const Alert = styled.div`
  padding: ${(props) => props.theme.spacings[2]}
    ${(props) => props.theme.spacings[3]};
  text-align: ${(props) => (props.textAlign ? props.textAlign : 'center')};
  background-color: ${getAlertBackgroundColor};
  color: ${getAlertTextColor};
  font-size: 1.25em;
`

export default Alert

Alert.propTypes = {
  /**
   * Type of alert, can be any type with a key in colors
   */
  alertType: PropTypes.string.isRequired,
  /**
   * Text alignment in the alert box
   */
  textAlign: PropTypes.string,
}

Alert.defaultProps = {
  textAlign: 'center',
}
