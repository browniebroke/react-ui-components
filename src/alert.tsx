import styled from 'styled-components'

import { getAlertBackgroundColor, getAlertTextColor } from './colors'
import { AlertOptions } from './types'

/**
 * Alert component
 */
const Alert = styled.div<AlertOptions>`
  padding: ${(props) => props.theme.spacings[2]}
    ${(props) => props.theme.spacings[3]};
  text-align: ${(props) => (props.textAlign ? props.textAlign : 'center')};
  background-color: ${getAlertBackgroundColor};
  color: ${getAlertTextColor};
  font-size: 1.25em;
`

export default Alert
