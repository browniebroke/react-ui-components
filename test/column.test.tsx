import React from 'react'
import renderer from 'react-test-renderer'

import { Col, styledComponentsTheme } from '../src'
import { ThemeProvider } from 'styled-components'

describe('Column component', () => {
  test('should render correctly without props', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with some props', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col xsWidth="50%" smWidth="33%" mdWidth="25%" lgWidth="20%" />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with textAlign', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col textAlign="center" />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with alignSelf', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col alignSelf="middle" />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with max width', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col
            MaxWidth={'100%'}
            xsMaxWidth="80%"
            smMaxWidth={'70%'}
            mdMaxWidth={`${(2 * 100) / 3}%`}
            lgMaxWidth="50%"
            xlMaxWidth="40%"
          />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with order props', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col
            xsOrder={'2'}
            smOrder={'3'}
            mdOrder={0}
            lgOrder={4}
            xlOrder={'6'}
          />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with global padding props', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col sidePadding={1} />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with each side padding props', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col
            topPadding={1}
            bottomPadding={2}
            rightPadding={3}
            leftPadding={4}
          />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with each direction padding props', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Col xPadding={1} yPadding={2} />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
