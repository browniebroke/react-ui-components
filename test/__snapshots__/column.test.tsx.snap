// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component should render correctly with alignSelf 1`] = `
<div
  className="sc-guDLRT dOIeyc"
/>
`;

exports[`Column component should render correctly with each direction padding props 1`] = `
<div
  className="sc-guDLRT uhlsv"
/>
`;

exports[`Column component should render correctly with each side padding props 1`] = `
<div
  className="sc-guDLRT kAPWbq"
/>
`;

exports[`Column component should render correctly with global padding props 1`] = `
<div
  className="sc-guDLRT efTRir"
/>
`;

exports[`Column component should render correctly with max width 1`] = `
<div
  className="sc-guDLRT cAeKDs"
/>
`;

exports[`Column component should render correctly with order props 1`] = `
<div
  className="sc-guDLRT PYIua"
/>
`;

exports[`Column component should render correctly with some props 1`] = `
<div
  className="sc-guDLRT omhJg"
/>
`;

exports[`Column component should render correctly with textAlign 1`] = `
<div
  className="sc-guDLRT cKbBAT"
/>
`;

exports[`Column component should render correctly without props 1`] = `
<div
  className="sc-guDLRT omhJg"
/>
`;
